.github/workflows/build.yml

```diff
@@ -37,16 +40,21 @@ jobs:
           sudo tar xf arm-gnu-toolchain-${ARM_GNU_TOOLCHAIN_VERSION}-x86_64-arm-none-eabi.tar.xz -C /opt
           echo "/opt/arm-gnu-toolchain-${ARM_GNU_TOOLCHAIN_VERSION}-x86_64-arm-none-eabi/bin" >> $GITHUB_PATH
 
-      - name: Build firmware
-        run: |
-          ./waf configure --board asterix_vla_dvb1 --nojs --nohash
-          ./waf build
-          ./waf build_prf
+      - name: Configure
+        run: ./waf configure --board ${{ matrix.board }} --nojs
+
+      - name: Build FW
+        run: ./waf build
+
+      - name: Build recovery FW
+        if: ${{ matrix.board == 'asterix_vla_dvb1' }}
+        run: ./waf build_prf
 
       - name: Store firmware images
         uses: actions/upload-artifact@v4
         with:
           name: firmware
+          overwrite: true
```

Comment:
question: why is this needed? can't all builds be uploaded?

Reply:
I misunderstood the failure in an earlier build, I thought it was trying to upload over a previous build and failing, but instead I just forgot to change the artefact name per board.